File: backend/src/main/java/edu/ucf/college_event_website/model/Event.java
package edu.ucf.college_event_website.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalTime;

@Getter
@Setter
@Entity
@Table(name = "events")
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "event_id")
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private LocalTime time;

    @Column(nullable = false)
    private LocalDate date;

    @ManyToOne
    @JoinColumn(name = "location_id", nullable = false)
    private Location location;

    @ManyToOne
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    @ManyToOne
    @JoinColumn(name="college_id", nullable = false)
    private College college;

    @Enumerated(EnumType.STRING)
    @Column(name = "event_type", nullable = false)
    private EventType eventType;

    @Column(name = "contact_phone", length = 20)
    private String contactPhone;

    @Column(name = "contact_email")
    private String contactEmail;
}

File: frontend/src/types/event.ts
import { Location } from "./location";

export type EventType = "PUBLIC" | "PRIVATE" | "RSO";

export interface Event {
  id: number;
  name: string;
  description: string;
  time: string;
  date: string;
  location: Location;
  createdBy: string;
  college: string;
  eventType: EventType;
  contactPhone: string;
  contactEmail: string;
  averageRating?: number;
  commentCount?: number;
  approved?: boolean; // For public events
  rsoId?: number; // For RSO events
}

export interface EventUpdateRequest {
  name: string;
  description: string;
  time: string;
  date: string;
  locationId: number;
  contactPhone: string;
  contactEmail: string;
}

export interface EventCreateRequest extends EventUpdateRequest {
  collegeId: number;
  eventType: EventType;
  rsoId?: number;
}

export interface EventComment {
  id: number;
  userId: number;
  content: string;
  date: string;
  time: string;
  userName: string;
}

export interface EventRating {
  eventId: number;
  averageRating: number;
  totalRatings: number;
}
